### PR TITLE
Update basic Manifold usage in the examples.

### PR DESCRIPTION
**Pull request by drwells**
I have updated most of the examples, but some need new pictures: I'll handle those later this week. This PR updates the examples that only need small changes.

### CODE REVIEW

File: examples/step-1/doc/results.dox [hunk 1]
```diff
@@ -53,11 +53,21 @@ Another possibility would be to generate meshes of entirely different
...
+GridGenerator namespace. Many of these geometries (such as the one used in this
+example program) contain cells with curved faces: put another way, we expect the
+new vertices placed on the boundary to lie along a circle. deal.II handles complex
+geometries with the Manifold class (and classes inheriting from it); in particular,
+the functions in GridGenerator corresponding to non-Cartesian grids (such as
+GridGenerator::hyper_shell or GridGenerator::truncated_cone) attach a Manifold
+object to the part of the triangulation that should be curved (SphericalManifold
+and CylindricalManifold, respectively) and use another manifold on the parts that
+should be flat (FlatManifold). See the documentation
+of Manifold or @ref manifolds module for descriptions of the design philosophy and
```
Comment (bangerth) [inline examples/step-1/doc/results.dox]: -> or THE @ref manifolds module

**Comment by drwells** [comment]
Part of #4800.

File: examples/step-1/doc/results.dox (hunk 1 above)
Comment (drwells) [reply examples/step-1/doc/results.dox]: Thanks. This is fixed.

**Comment by drwells** [comment]
@bangerth Yes; its very nice that we can delete all of this explanatory text and now just write "GridGenerator did the right thing so lets move on"

**Comment by drwells** [comment]
Per @bangerth's review I will merge this now.

**Pull request merged**